core/src/main/java/feign/FeignException.java

```diff
@@ -37,108 +37,61 @@
 
   private static final String EXCEPTION_MESSAGE_TEMPLATE_NULL_REQUEST =
       "request should not be null";
+  private static final String EXCEPTION_MESSAGE_TEMPLATE_NULL_RESPONSE =
+      "response should not be null";
   private static final long serialVersionUID = 0;
-  private int status;
-  private byte[] responseBody;
-  private Map<String, Collection<String>> responseHeaders;
+  private final int status;
   private Request request;
-
-  protected FeignException(int status, String message, Throwable cause) {
```

Comment:
I HATE this class!

That said, it's not safe to remove the extra constructors, as people may be using it.

Bring them back, and feel free to deprecate them all

Reply:
done in the commit 2523ef1